app/src/main/java/com/startup/go4lunch/ui/RestaurantListAdapter.java

```diff
@@ -0,0 +1,60 @@
+package com.startup.go4lunch.ui;
+
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.TextView;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+import androidx.recyclerview.widget.DiffUtil;
+import androidx.recyclerview.widget.ListAdapter;
+import androidx.recyclerview.widget.RecyclerView;
+
+import com.startup.go4lunch.R;
+import com.startup.go4lunch.model.Restaurant;
+
+import java.util.List;
+
+public class RestaurantListAdapter extends ListAdapter<Restaurant, RestaurantListAdapter.ViewHolder> {
+
+    protected RestaurantListAdapter(@NonNull DiffUtil.ItemCallback<Restaurant> diffCallback) {
+        super(diffCallback);
+    }
+
+    @Override
+    public void submitList(@Nullable List<Restaurant> list) {
+        super.submitList(list);
+    }
+
+    @NonNull
+    @Override
+    public ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
+        View view = LayoutInflater.from(parent.getContext())
+                .inflate(R.layout.item_restaurant, parent, false);
+        return new ViewHolder(view);
+    }
+
+    @Override
+    public void onBindViewHolder(@NonNull ViewHolder holder, int position) {
+        holder.restaurantName.setText(getCurrentList().get(position).getName());
```

Comment:
Extract getCurrentList().get(position) in a local variable

Reply:
Ok